shell/app/config-page/components/bar-chart/bar-chart.tsx

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// This program is free software: you can use, redistribute, and/or modify
+// it under the terms of the GNU Affero General Public License, version 3
+// or later ("AGPL"), as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import React from 'react';
+import EChart from 'charts/components/echarts';
+import { newColorMap, getClass } from 'config-page/utils';
+import { EmptyHolder } from 'common';
+import { map, uniq, merge, get } from 'lodash';
+import theme from 'app/themes/theme';
+import './bar-chart.scss';
+
+const handleAxis = (opt: Obj, yAxisLabelLen = 10) => {
+  const reOpt = { ...opt };
+  if (opt.yAxis) {
+    const adjustYAxis = (_axis: Obj) => {
+      return {
+        ..._axis,
+        axisLabel: {
+          formatter: (v: string) => (v.length > yAxisLabelLen ? `${v.substr(0, yAxisLabelLen)}...` : v),
+          color: theme.chart.yAxisColor,
+          ...(_axis.axisLabel || {}),
+        },
+      };
+    };
+
+    reOpt.yAxis = Array.isArray(reOpt.yAxis) ? reOpt.yAxis.map(adjustYAxis) : adjustYAxis(reOpt.yAxis);
```

Comment:
`adjustYAxis(reOpt.yAxis)` will not modify reOpt.yAxis, return value is not used.